File: dotnet/src/Model/Enemy/Buildings/EGasCollector.cs
using System;

namespace PlanetCS1 {
    class EGasCollector : GasCollector {
        public EGasCollector(Location location) {
            this.loc = location;
            this.use = true;
            this.owner = 2;
            this.icon = "eGC";
        }
    }
}

File: typescript/src/PlanetTS1/Model/Enemy/Buildings/EGasCollector.ts
import { ALocation as ALocation } from "../../Utilities"

export class EGasCollector extends GasCollector {
    loc : ALocation
    use : boolean
    owner : number
    icon : string
    constructor(location : ALocation) {
        super()
        this.loc = location
        this.use = true
        this.owner = 2
        this.icon = "eGC"
    }
}